### PR TITLE
feat: add `_tools/eslint/rules/line-closing-bracket-spacing`

### PR DESCRIPTION
**Pull request by headlessNode**
## Description

> What is the purpose of this pull request?

This pull request:

-   add `eslint/rules/line-closing-bracket-spacing`
-   add `fixer` function for `eslint/rules/line-closing-bracket-spacing`

## Related Issues

> Does this pull request have any related issues?

No. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: etc/eslint/rules/stdlib.js [hunk 1]
```diff
@@ -346,8 +346,7 @@ rules[ 'stdlib/jsdoc-checkbox-character-style' ] = [
 	{
 		'checked': 'x',
 		'unchecked': ' '
-	}
-];
+	}];
```
Comment (kgryte) [inline etc/eslint/rules/stdlib.js]: @headlessNode Why these changes? The existing code should be fine.

File: etc/eslint/rules/stdlib.js [hunk 2]
```diff
@@ -502,8 +501,7 @@ rules[ 'stdlib/jsdoc-code-block-style' ] = [ 'error', 'fenced' ];
 rules[ 'stdlib/jsdoc-fenced-code-flag' ] = [
 	'error', {
 		'allowEmpty': false
-	}
-];
+	}];
```
Comment (kgryte) [inline etc/eslint/rules/stdlib.js]: Same question.

File: etc/eslint/rules/stdlib.js [hunk 3]
```diff
@@ -1649,8 +1647,7 @@ rules[ 'stdlib/jsdoc-list-item-spacing' ] = 'error';
 rules[ 'stdlib/jsdoc-markdown-remark' ] = [ 'error',
 	{
 		'config': require( './../../remark/.remarkrc.jsdoc.js' )
-	}
-];
+	}];
```
Comment (kgryte) [inline etc/eslint/rules/stdlib.js]: Same question.

**Review by kgryte — CHANGES_REQUESTED** [review]
Other than the style changes, this PR LGTM.

File: etc/eslint/rules/stdlib.js (hunk 1 above)
Comment (headlessNode) [reply etc/eslint/rules/stdlib.js]: @kgryte The rule is taking the new-line between closing brackets as a space. Which should be the correct behavior, no?

Same thing happens with `eol-open-bracket-spacing` rule.

We don't allow the following:
```
var arr = [
    {
       'a': 1
    }
];
```

It should be:
```
var arr = [{
    'a': 1
}];
```

Comment (kgryte) [reply etc/eslint/rules/stdlib.js]: Ah! The first one should be allowed for `eol-open-bracket-spacing`. The only thing we want to disallow is when someone does 

```
foo = [ {
    ...
} ];
```

That is it. The following should be perfectly valid.

```
foo = [{
    ...
}];

foo = [
    {
        ...
    }
];

foo = [ {...} ];
```

Comment (headlessNode) [reply etc/eslint/rules/stdlib.js]: @kgryte Makes sense. Thanks!

**Comment by headlessNode** [comment]
@kgryte Changes applied. I'll add similar changes to `eol-open-bracket-spacing` in a separate PR.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @headlessNode!
